code/__HELPERS/game.dm

```diff
@@ -42,8 +42,20 @@
 
 	var/lum = source.luminosity
 	source.luminosity = world.view
-
 	var/list/heard = view(range, source)
+	for(var/ear in view(range+3, source))
+		if(!ear)
+			continue
+		if(ear in heard)
+			continue
+		if(!ishuman(ear))
+			continue
+		var/mob/living/carbon/human/H = ear
+		if(get_dist(H.loc, source) < range)
```

Comment:
You're kinda skipping the perk check here. It's continuing if somebody is within range before checking if they have the perk.

Reply:
Yes, because I'm only interested in humans who are in a range between `[range]` and `[range + 3]`. Those who are below [range] should already be on the `heard ` list.
![image](https://user-images.githubusercontent.com/49106866/85955972-4129c480-b950-11ea-8518-cbc11180c5a5.png)

Reply:
Then why not subtract heard from view(range+3, source)?

Reply:
Something like: 
```
heard2 = view(range+3, source) - heard
    for(i in heard2) 
      if(perk)
          heard += i
```?

Reply:
Basically